### PR TITLE
Run Control Services

### PR DESCRIPTION
**Pull request by TiagoTAlves**
Pods intialise as we expect but run services (both run registry and run number) database does not work

### CODE REVIEW

**Comment by jcpunk** [comment]
Ideally, I'd like to have the schema's official copy in the microservices repo along side the code that's using it.  That way if the usage changes it is easy to update both at once.

**Comment by TiagoTAlves** [comment]
Both run number and run registry work as expected now. Will give a series of tests for reviewers to make sure everything works

File: manifests/postgres/sql/create-run-databases.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e
+
+psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" --dbname "$POSTGRES_DB" <<-EOSQL
```
Comment (jcpunk) [inline manifests/postgres/sql/create-run-databases.sh]: Can you put the schema into a separate file that is pulled into this script?  The maintained helm charts are able to read in a raw sql file to generate the initial schema and I'd like to use that logic long term.

File: manifests/microservices/runnumber-rest.yaml [hunk 2]
```diff
@@ -7,14 +7,17 @@ spec:
...
       labels:
         app: runnumber-rest
     spec:
+      initContainers:
+      - name: wait-for-run-postgres
+        image: busybox:1.28
+        command: ['sh','-c', 'until nslookup runnumber-rest-svc-ext.microservices.svc.cluster.local; do echo waiting for run-postgres to run; sleep 1; done']            
       containers:
-      - image: ghcr.io/dune-daq/pocket-runnumber-rest:0.8
+      - image: ghcr.io/dune-daq/pocket-runnumber-rest:1.0
```
Comment (jcpunk) [inline manifests/microservices/runnumber-rest.yaml:20]: I'm not showing any code changes pending up at https://github.com/DUNE-DAQ/microservices to track the container tag changes...

Comment (TiagoTAlves) [reply manifests/microservices/runnumber-rest.yaml:20]: I'm not on the DUNE DAQ microservices group so I couldn't upload my changes on to microservices to track the container changes.

Comment (plasorak) [reply manifests/microservices/runnumber-rest.yaml:20]: @TiagoTAlves , can you fork it, create a PR on microservices and advertise it here?

Comment (jcpunk) [reply manifests/microservices/runnumber-rest.yaml:20]: I'm showing https://github.com/DUNE-DAQ/microservices/tree/titavare/run-number exists now.  I pushed a few additional changes into that branch so the containers can run as non-root.

Long term I'd like to store the kubernetes yaml for the service with the code that builds the container.  That way if things change further down the line the deployments are easy to sync up with specific container identities.

File: manifests/postgres/sql/create-run-databases.sh (hunk 1 above)
Comment (plasorak) [reply manifests/postgres/sql/create-run-databases.sh]: > raw sql file

Unfortunately, this isn't a raw sql script, for example, `\c` is a way to change the database, but we haven't figured out how to do this in raw sql.

File: manifests/postgres/sql/create-run-databases.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+);
+
+CREATE TABLE run_schema.run_number(
+  rn INT PRIMARY KEY NOT NULL,
+  start_time TIMESTAMP (6) NOT NULL,
+  flag BOOLEAN NOT NULL,
+  stop_time TIMESTAMP (6)
+);
+
+INSERT INTO run_schema.run_number(rn, start_time, flag, stop_time) VALUES (1000000, CURRENT_TIMESTAMP, true, CURRENT_TIMESTAMP);
```
Comment (plasorak) [inline manifests/postgres/sql/create-run-databases.sh]: It occurs to me that it'd be nice to have the first run number to be configurable. Maybe this isn't for this PR, but a nice-to-have?

Comment (TiagoTAlves) [reply manifests/postgres/sql/create-run-databases.sh]: We are no longer using this bash script right now unless I figure out how to run the the sql file as a psql command in the bash script.

Comment (TiagoTAlves) [reply manifests/postgres/sql/create-run-databases.sh]: Right now we are not inserting any data into the runnumber db via the configmap

File: manifests/postgres/sql/runservices.sql [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+SET default_tablespace = '';
+
+SET default_table_access_method = heap;
+
+--
+-- Name: run_number; Type: TABLE; Schema: run_schema; Owner: admin
+--
+
+CREATE TABLE run_schema.run_number (
```
Comment (plasorak) [inline manifests/postgres/sql/runservices.sql:43]: I think this table still misses the first entry for run number, at least I cannot see it.

Comment (plasorak) [reply manifests/postgres/sql/runservices.sql:43]: I mean the INSERT statement, with the first run number we want to use.

File: manifests/postgres/sql/create-run-databases.sh (hunk 3 above)
Comment (plasorak) [reply manifests/postgres/sql/create-run-databases.sh]: If we are not using this file, please delete it.

**Comment by TiagoTAlves** [comment]
To test:

**Comment by TiagoTAlves** [comment]
To Test:
Run pocket with only run services `$ SERVICES=runserv make setup.local`
(If kubectl does not work dont forget to use ` eval $(make env)`)
Wait for all pods to load, you can check status with `$ kubectl get pods -A`
Test to see if database is working as expected using some curl commands:

RunRegistry
` $ curl -u fooUsr:barPass -X GET np04-srv-019:30005/runregistry/getRunMeta/2`
` $ curl -u fooUsr:barPass -X GET -O -J np04-srv-019:30005/runregistry/getRunBlob/2`

RunNumber:
` $ curl -u fooUsr:barPass http://np04-srv-019:30006/runnumber/get `
` $ curl -u fooUsr:barPass http://np04-srv-019:30006/runnumber/get `

File: manifests/postgres/sql/create-run-databases.sh (hunk 1 above)
Comment (TiagoTAlves) [reply manifests/postgres/sql/create-run-databases.sh]: These commits d1eab9f075c150e171a831ef804a6dbb9fea42be & 0a6411ab5a241072df8633a660540f5dcf424bee solve this issue

File: manifests/postgres/sql/create-run-databases.sh (hunk 3 above)
Comment (TiagoTAlves) [reply manifests/postgres/sql/create-run-databases.sh]: Addressed here 0a6411ab5a241072df8633a660540f5dcf424bee

**Review by jcpunk — APPROVED** [review]
This seems reasonable to me.  I'd like to see the pods run as a non-root user, but that isn't blocking in my opinion.

**Pull request merged**